src/qcodes/dataset/data_set_protocol.py

```diff
@@ -276,6 +283,51 @@ def the_same_dataset_as(self, other: DataSetProtocol) -> bool: ...
 class BaseDataSet(DataSetProtocol, Protocol):
     # shared methods between all implementations of the dataset
 
+    def _collect_all_related_parameters(
```

Comment:
I feel like this entire method should live in the `InterDependencies_` class, absent the check against the `result_dict` keys, which can easily be done in `_enqueue_results` from the returned list of all related parameters